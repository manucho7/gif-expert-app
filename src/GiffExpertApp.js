import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GiffExpertApp = ({ defaultCategories= [] }) => {
  //usamos hooks
  //const [categories, setCategories] = useState(['One Punch']);
  const [categories, setCategories] = useState( defaultCategories );

    return (
        <>
            <h2>GiffExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {
                    categories.map((category) => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    );
};

export default GiffExpertApp;
