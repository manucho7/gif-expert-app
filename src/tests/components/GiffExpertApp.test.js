import '@testing-library/jest-dom';
import '@testing-library/dom';
import { shallow } from 'enzyme';
import GiffExpertApp from '../../GiffExpertApp';
import React from 'react';


describe('Pruebas en <GiffExpertApp />', () => {
    
    test('debe mostrarse correctamente', () => {
        
        const wrapper = shallow( <GiffExpertApp /> );
        expect( wrapper ).toMatchSnapshot();

    });
    
    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GiffExpertApp defaultCategories={ categories }/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    

})
