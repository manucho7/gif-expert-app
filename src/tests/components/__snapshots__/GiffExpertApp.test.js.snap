// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiffExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiffExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GiffExpertApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GiffExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
